swap/swap_test.go

```diff
@@ -1508,3 +1565,143 @@ func TestAvailableBalance(t *testing.T) {
 	}
 
 }
+func TestBouncedCheque(t *testing.T) {
+	testBackend := newTestBackend(t)
+	defer testBackend.Close()
+	// create both test swap accounts
+	creditorSwap, clean1 := newTestSwap(t, beneficiaryKey, testBackend)
+	debitorSwap, clean2 := newTestSwap(t, ownerKey, testBackend)
+	defer clean1()
+	defer clean2()
+
+	testAmount := DefaultPaymentThreshold + 42
+
+	ctx := context.Background()
+	err := testDeploy(ctx, creditorSwap, int256.Uint256From(0))
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = testDeploy(ctx, debitorSwap, int256.Uint256From(testAmount))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// create Peer instances
+	// NOTE: remember that these are peer instances representing each **a model of the remote peer** for every local node
+	// so creditor is the model of the remote mode for the debitor! (and vice versa)
+	cPeer := newDummyPeerWithSpec(Spec)
+	dPeer := newDummyPeerWithSpec(Spec)
+	creditor, err := debitorSwap.addPeer(cPeer.Peer, creditorSwap.owner.address, debitorSwap.GetParams().ContractAddress)
+	if err != nil {
+		// ON CONNECT TO PEER NO BOUNCED CHEQUE
+		if err == ErrBouncedCheque {
+			t.Fatalf("Bounced cheque in chequebook")
+		} else {
+			t.Fatal(err)
+		}
+	}
+
+	debitor, err := creditorSwap.addPeer(dPeer.Peer, debitorSwap.owner.address, debitorSwap.GetParams().ContractAddress)
+	if err != nil {
+		// ON CONNECT TO PEER NO BOUNCED CHEQUE
```

Comment:
please format the comments here correctly and make them a bit more readable if possible